docs/docs/recipes/integrate-logto/express.mdx

```diff
@@ -0,0 +1,167 @@
+
+<AppNote type="Traditional web" />
+
+## Add dependencies
+
+<Tabs>
+
+  <TabItem value="npm" label="npm">
+
+```bash
+npm i @logto/express cookie-parser express-session
+```
+
+  </TabItem>
+  <TabItem value="yarn" label="Yarn">
+
+```bash
+yarn add @logto/express cookie-parser express-session
+```
+
+  </TabItem>
+  <TabItem value="pnpm" label="pnpm">
+
+```bash
+pnpm add @logto/express cookie-parser express-session
+```
+
+  </TabItem>
+
+</Tabs>
+
+## Init LogtoClient
+
+<AssumingUrl />
+
+<GetAppSecret />
+
+Import and initialize LogtoClient:
+
+```ts
+import { LogtoExpressConfig } from '@logto/express';
+
+const config: LogtoExpressConfig = {
+  appId: '<your-application-id>',
+  appSecret: '<your-app-secret-copied-from-console>',
+  endpoint: '<your-logto-endpoint>', // E.g. http://localhost:3001
+  baseUrl: '<your-express-app-base-url>', // E.g. http://localhost:3000
+};
+```
+
+### Prepare required middlewares
+
+The SDK requires that [express-session](https://www.npmjs.com/package/express-session) is configured first.
+
+```ts
+import cookieParser from 'cookie-parser';
+import session from 'express-session';
+
+app.use(cookieParser());
+app.use(session({ secret: 'keyboard cat', cookie: { maxAge: 14 * 24 * 60 * 60 } }));
+```
+
+## Sign in
+
+<SignInFlowSummary />
+
+### Configure sign-in redirect URI
+
+<ConfigureRedirectUri
+  figureSrc={redirectUriFigure}
+  redirectUri="http://localhost:3000/logto/sign-in-callback"
+/>
+
+### Prepare logto routes
+
+Prepare routes to connect with Logto.
+
+Go back to your IDE/editor, use the following code to implement the API routes first:
+
+```ts
+import { handleAuthRoutes } from '@logto/express';
+
+app.use(handleAuthRoutes(config));
+```
+
+This will create 3 routes automatically:
+
+1. `/logto/sign-in`: Sign in with Logto.
+2. `/logto/sign-in-callback`: Handle sign-in callback.
+3. `/logto/sign-out`: Sign out with Logto.
+
+### Implement sign-in
+
+We're almost there! In order to sign in, we can just redirect the user to Logto's sign-in route. For example, create a sign in link button:
+
+```ts
+app.get('/', (req, res) => {
+  res.setHeader('content-type', 'text/html');
+  res.end(`<div><a href="/logto/sign-in">Sign In</a></div>`);
+});
+```
+
+## Get user profile
+
+To get user profile, we need to add another middleware:
+
+```ts
+import { withLogto } from '@logto/express';
+
+app.use(withLogto(config));
+```
+
+Then the user profile will be attached to `req`, example usage:
+
+```ts
+app.get('/user', (req, res) => {
+  res.json(req.user);
+});
+```
+
+## Protect routes
+
+After setting up `withLogto` in the previous step, we can protect routes by creating a simple middleware:
+
+```ts
+const requireAuth = async (req: Request, res: Response, next: NextFunction) => {
+  if (!req.user.isAuthenticated) {
+    res.redirect('/logto/sign-in');
+  }
+
+  next();
+};
+```
+
+And then:
+
+```ts
+app.get('/protected', requireAuth, (req, res) => {
+  res.end('protected resource');
+});
+```
```

Comment:
nit: use `request` and `response` since that's our JS code convention.

Reply:
ditto